packages/@headlessui-react/src/components/radio-group/radio-group.tsx

```diff
@@ -315,7 +315,8 @@ function RadioGroupFn<TTag extends ElementType = typeof DEFAULT_RADIO_GROUP_TAG,
             {name != null && (
               <FormFields
                 disabled={disabled}
-                data={value != null ? { [name]: value || 'on' } : {}}
+                data={{ [name]: value || 'on' }}
```

Comment:
I know it doesn't technically matter but it seems weird to me that this produces a hidden radio input where the value="on" (it's not checked so it's not going to show up in form data).

I feel like I would've expected to see three radios in the DOM with the same name, the three values, and only the appropriate one checked. Instead of mutating a single radio.

Maybe it doesn't matter though? Thoughts?

Reply:
It's a bit odd for sure, but if you submit a form with `<input type="radio" name="foo" />` without a value, then it will be submitted with a value of `"on"`.

We could try to move the hidden inputs to each `Radio` component itself, but only the checked ones will be submitted anyway so this feels better. Then we don't have to render `n` hidden radio inputs.